lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -70,17 +76,37 @@ protected void setup(Binder binder)
         else {
             install(new HiveS3Module());
         }
+
+        newMapBinder(binder, String.class, TrinoFileSystemCache.class);
+        if (config.getCacheType() != null) {
+            install(conditionalModule(
+                    FileSystemConfig.class,
+                    cache -> "alluxio".equalsIgnoreCase(cache.getCacheType()),
+                    new AlluxioFileSystemCacheModule()));
+        }
+        else {
+            binder.bind(NodeProvider.class).to(NoneNodeProvider.class).in(SINGLETON);
+        }
     }
 
     @Provides
     @Singleton
     public TrinoFileSystemFactory createFileSystemFactory(
             HdfsFileSystemFactoryHolder hdfsFileSystemFactory,
             Map<String, TrinoFileSystemFactory> factories,
+            Optional<TrinoFileSystemCache> fileSystemCache,
             Tracer tracer)
     {
         TrinoFileSystemFactory delegate = new SwitchingFileSystemFactory(hdfsFileSystemFactory.value(), factories);
-        return new TracingFileSystemFactory(tracer, delegate);
+        return new TracingFileSystemFactory(tracer,
+                fileSystemCache.map(cache -> (TrinoFileSystemFactory) new CacheFileSystemFactory(delegate, cache)).orElse(delegate));
+    }
+
+    @Provides
+    @Singleton
+    public Optional<TrinoFileSystemCache> createFileSystemCache(FileSystemConfig config, Map<String, TrinoFileSystemCache> caches)
+    {
+        return Optional.ofNullable(caches.get(config.getCacheType()));
```

Comment:
We should fail here if `cacheType` is configured (non-null) but no cache is found